lib/promscrape/discovery/kubernetes/api_watcher.go

```diff
@@ -579,6 +579,9 @@ func (uw *urlWatcher) reloadObjects() string {
 		return uw.resourceVersion
 	}
 
+	// set by default to zero value, it reduces a load for kubernetes control plane
+	// https://kubernetes.io/docs/reference/using-api/api-concepts/#semantics-for-watch
```

Comment:
Should the link be https://kubernetes.io/docs/reference/using-api/api-concepts/#semantics-for-get-and-list? 
Since we are performing `list` instead of `watch` request here.
>Most recent
Return data at the most recent resource version. The returned data must be consistent (in detail: served from etcd via a quorum read).

Reply:
Thanks! Addressed this in the updated pull request